src/ucp/core/ucp_mm.c

```diff
@@ -437,6 +437,32 @@ static void ucp_memh_cleanup(ucp_context_h context, ucp_mem_h memh)
     ucs_free(memh);
 }
 
+ucs_status_t ucp_memh_register_one(ucp_context_h context, ucp_mem_h memh,
```

Comment:
I would prefer to have something like `ucp_memh_check_alignment` which would just adjust address and length. This function should be called inside `ucp_memh_register` before `uct_md_mem_reg_v2`. WDYT?

Reply:
Would it looks ugly if we save both aligned and original pointers on the `ucp_memh_register` stack and then align only the required part. E.g.:

```C
ucp_memh_register(...) {
    void *address     = ucp_memh_address(memh);
    size_t length     = ucp_memh_length(memh);
    void *reg_address = address;
    size_t reg_length = length;

...

    ucp_memh_adjust_alignment(&reg_address, &reg_length);

...

    return uct_md_mem_reg_v2(context->tl_mds[md_index].md, reg_address,
                             reg_length, reg_params, &memh->uct[md_index]);
}

```

Even with the current approach I want to propose to rename `ucp_memh_register_one` to `ucp_memh_reg_align` or `ucp_memh_register_internal`. WDYT?

Reply:
`ucp_memh_reg_align` looks good, but then better to use `ucp_memh_register_align[ed]` for consistency